sky/provision/kubernetes/utils.py

```diff
@@ -131,6 +135,17 @@ def wrapper(*args, **kwargs):
             backoff = common_utils.Backoff(initial_backoff=retry_interval,
                                            max_backoff_factor=3)
 
+            # try to capture the kubernetes context from parameters
+            if context_param:
+                context = kwargs.get(context_param)
+                if context is None and len(args) > 0:
+                    sig = inspect.signature(func)
```

Comment:
This is quite hacky. How about we just enforce `context` to be appear in kwargs, and all wrapped function to take `context` as kwargs? `def wrapped_function(myarg1, *, context):`

Reply:
Sounds good, is there any sound solution to enforce `context` to be appear in kwargs?

Reply:
How about we just `assert` it in the decorator function, which will only be triggerred when the function is being called, but it might be fine.